gateway/api/guardrails/guardrails.go

```diff
@@ -86,20 +120,52 @@ func Put(c *gin.Context) {
 		Output:      req.Output,
 		UpdatedAt:   time.Now().UTC(),
 	}
+
 	err := models.UpdateGuardRailRules(rule)
 	switch err {
 	case models.ErrNotFound:
 		c.JSON(http.StatusNotFound, gin.H{"message": err.Error()})
 		return
 	case nil:
+		// Sync connections
+		if len(req.ConnectionIDs) > 0 {
```

Comment:
The problem with the sync approach and not having this in the core methods (Update, Create, List) it could return an empty list of connection ids in case it fails. This process is turning this endpoint inconsistent and it could be difficult to troubleshoot these kind of errors.

I suggest to refactor to perform all this operation in a single transaction, however I leave to you to decide if you want to merge considering these shortcomings to improve in the future.